### PR TITLE
Updating react router

### PR DESCRIPTION
**Pull request by deadlybutter**
This PR updates the react router dependency to the latest version & does some minor fixes to make that work!

- Updating some relevant peer dependencies 
- Removing usage of `syncHistoryWithStore` because it was [removed](https://github.com/ReactTraining/react-router/issues/4869).

### CODE REVIEW

**Comment by weerd** [comment]
@deadlybutter looks good, but lets have @DFurnes 👁  it as well, just in case :)

File: package.json [hunk 1]
```diff
@@ -31,16 +31,16 @@
...
     "marked": "^0.3.6",
     "prop-types": "^15.5.8",
     "react": "^15.4.2",
     "react-dom": "^15.4.2",
     "react-redux": "^5.0.2",
-    "react-router": "^3.0.2",
+    "react-router": "^4.5.2",
     "react-router-dom": "^4.0.0-beta.6",
-    "react-router-redux": "^4.0.8",
+    "react-router-redux": "^5.0.0-alpha.6",
```
Comment (DFurnes) [inline package.json]: Let's pin to this exact release since it's an alpha.

File: resources/assets/store.js [hunk 2]
```diff
@@ -70,8 +70,8 @@ const initialState = {
  * @param preloadedState
  * @returns {Store<S>}
  */
-export function configureStore(reducers, preloadedState = {}) {
-  const middleware = [thunk, observerMiddleware];
+export function configureStore(reducers, initialMiddleware, preloadedState = {}) {
+  const middleware = [thunk, observerMiddleware, ...initialMiddleware];
```
Comment (DFurnes) [inline resources/assets/store.js]: Should we just pass in all the middleware from `init.js`?

**Comment by DFurnes** [comment]
Build is failing. Sure you have the right React-Router version?

```
npm ERR! notarget No compatible version found: react-router@^4.5.2
npm ERR! notarget Valid install targets:
npm ERR! notarget 4.1.1, 4.1.0, 4.0.0, 4.0.0-beta.8, 4.0.0-beta.7, 4.0.0-beta.6, 4.0.0-beta.5, 4.0.0-beta.4, 4.0.0-beta.3, 4.0.0-beta.2, 4.0.0-beta.1, 4.0.0-alpha.6, 4.0.0-alpha.5, 4.0.0-alpha.4, 4.0.0-alpha.3, 4.0.0-2, 4.0.0-1, 4.0.0-0, 3.0.5, 3.0.4, 3.0.3, 3.0.2, 3.0.1, 3.0.0, 3.0.0-beta.1, 3.0.0-alpha.3, 3.0.0-alpha.2, 3.0.0-alpha.1, 2.8.1, 2.8.0, 2.7.0, 2.6.1, 2.6.0, 2.5.2, 2.5.1, 2.5.0, 2.4.1, 2.4.0, 2.3.0, 2.2.4, 2.2.2, 2.2.1, 2.2.0, 2.1.1, 2.1.0, 2.0.1, 2.0.0, 2.0.0-rc6, 2.0.0-rc5, 2.0.0-rc4, 2.0.0-rc3, 2.0.0-rc2, 2.0.0-rc1, 1.0.3, 1.0.2, 1.0.1, 1.0.0, 1.0.0-rc4, 1.0.0-rc3, 1.0.0-rc2, 1.0.0-rc1, 1.0.0-beta4, 1.0.0-beta3, 1.0.0-beta2, 1.0.0-beta1, 1.0.0-alpha2, 1.0.0-alpha1, 0.13.6, 0.13.5, 0.13.4, 0.13.3, 0.13.2, 0.13.1, 0.13.0, 0.12.4, 0.12.3, 0.12.2, 0.12.1, 0.12.0, 0.11.6, 0.11.5, 0.11.4, 0.11.3, 0.11.2, 0.11.1, 0.11.0, 0.10.2, 0.10.1, 0.10.0, 0.9.5, 0.9.4, 0.9.3, 0.9.2, 0.9.1, 0.9.0, 0.8.0, 0.7.0, 0.6.1, 0.6.0, 0.5.3, 0.5.2, 0.5.1, 0.5.0, 0.4.2, 0.4.1, 0.4.0, 0.0.1, 0.0.0
npm ERR! notarget 
npm ERR! notarget This is most likely not a problem with npm itself.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

**Comment by deadlybutter** [comment]
That's weird, let me wipe my npm modules & double check all that

**Comment by deadlybutter** [comment]
Well this turned out to be more annoying than I expected 

- Fixed react router version
- Fixed History usage to be up to date & changed how its used
- Fixed App component Router type & removing children routes usage (`Warning: You should not use <Route component> and <Route children> in the same route; <Route children> will be ignored
warning @ browser.js:49`)
- Fixed wrong import (wanted to break the computer over [this one](https://github.com/ReactTraining/react-router/issues/4477#issuecomment-300257177))
- Fixed active class name not being a thing (Has to be surrounded by `connect()` and that seemed like more work than its worth)
- Fixed wrapper prop type error
- Moved all middleware to the init file

**Comment by weerd** [comment]
> Fixed wrong import (wanted to break the computer over this one)
@deadlybutter I had a similar issue a while back and was ready to toss my computer @_@

File: resources/assets/containers/NavigationContainer.js [hunk 3]
```diff
@@ -5,27 +5,32 @@ import { Navigation, NavigationLink } from '../components/Navigation';
...
+        {page.fields.title}
+      </NavigationLink>
+    );
+  });
 
   return (
     <Navigation>
-      <NavigationLink to="/">Community</NavigationLink>
-      <NavigationLink to="/action">Action</NavigationLink>
+      <NavigationLink to="/" active={pathname === '/'}>Community</NavigationLink>
```
Comment (DFurnes) [inline resources/assets/containers/NavigationContainer.js]: Can you use [`NavLink`](https://reacttraining.com/react-router/web/api/NavLink) so we don't have to do this manually for every link?

Comment (deadlybutter) [inline resources/assets/containers/NavigationContainer.js]: @DFurnes This is using NavLink, the URL is /action but the highlighted portion & content is for the community.

screen shot 2017-05-11 at 5 01 31 pm

```js
      <NavLink className="nav-link" to="/" activeClassName="is-active">Community</NavLink>
      <NavLink className="nav-link" to="/action" activeClassName="is-active">Action</NavLink>
```

**Comment by DFurnes** [comment]
@deadlybutter I just pushed up some commits that get this ready to merge. The two main changes were adding the `withRouter()` higher-order component around the Chrome container so it updates when the route changes, and importing `Route` from the correct package (was undefined).

**Comment by deadlybutter** [comment]
@DFurnes Github won't let me approve (Its my PR 😆 ) but these commits seem fine to me. Thank you for digging through that...

**Comment by DFurnes** [comment]
Whoops, two more fixes: by default `NavLink` will "fuzzy" match so community link was always selected, and added a `<Switch>` so we don't always show the 404 route at the bottom of the page.

**Pull request merged**